larpandora/LArPandoraInterface/LArPandoraInput.cxx

```diff
@@ -215,17 +215,53 @@ void LArPandoraInput::CreatePandoraLArTPCs(const Settings &settings, const LArDr
 
 void LArPandoraInput::CreatePandoraDetectorGaps(const Settings &settings, const LArDriftVolumeList &driftVolumeList, const LArDetectorGapList &listOfGaps)
 {
+    //ATTN - Unlike SP, DP detector gaps are not in the drift direction 
+    art::ServiceHandle<geo::Geometry const> theGeometry;
+    const bool isDualPhase(theGeometry->MaxPlanes() == 2);
+    
     mf::LogDebug("LArPandora") << " *** LArPandoraInput::CreatePandoraDetectorGaps(...) *** " << std::endl;
-
+    
     if (!settings.m_pPrimaryPandora)
         throw cet::exception("LArPandora") << "CreatePandoraDetectorGaps - primary Pandora instance does not exist ";
-
+    
     const pandora::Pandora *pPandora(settings.m_pPrimaryPandora);
-
+    
     for (const LArDetectorGap &gap : listOfGaps)
     {
         PandoraApi::Geometry::LineGap::Parameters parameters;
-
+        
+        if (isDualPhase)
+        {
+            const float gapSizeY (std::fabs(gap.GetY2() - gap.GetY1())); //Could have chosen Z here, resulting in switching Y<->Z and U<->V in the try{...} block below
+            
+            const float maxGapSize (30.); //Match the maxDisplacement variable hard-coded value in LArPandoraGeometry
```

Comment:
For clarity, could use gapSizeY and maxGapSize to form a const Boolean, which is what is actually used below, in the assignment of parameters. Can collect the comments on the three lines here and make a compact but comprehensive note above the definition of the Boolean.

Are there any limitations to when this gapSize vs. maxGapSize check works, or is it completely reliable?

We should make a note to follow up any hard-coded values!